app/src/main/java/com/example/healthinspector/Fragments/ScanFlow/ScanFragment.java

```diff
@@ -0,0 +1,181 @@
+                                    }
+                                    for(int i = 0; i < ingredientsAnalysis.size(); i++){
+                                        ingredientsAnalysis.set(i, ingredientsAnalysis.get(i).substring(3).replace("-", " "));
+                                    }
+                                    if(response.getJSONObject(getString(R.string.product)).has(getString(R.string.nova_group))){
+                                        int novaGroupNumber = response.getJSONObject(getString(R.string.product)).getInt(getString(R.string.nova_group));
+                                        novaGroup = novaGroups.get(novaGroupNumber);
+                                    }
+
+                                    ArrayList<String> nutrientLevels = new ArrayList<>();
+                                    JSONObject nutrientLevelsJSON = response.getJSONObject(getString(R.string.product)).getJSONObject(getString(R.string.nutrient_levels));
+                                    Iterator iterator =  nutrientLevelsJSON.keys();
+                                    while (iterator.hasNext()){
+                                        String key = iterator.next().toString();
+                                        nutrientLevels.add(key + ": " + nutrientLevelsJSON.getString(key));
+                                    }
+                                    ScannedProduct scannedProduct = new ScannedProduct(productName, healthInspectorScore, ingredients, ingredientsAnalysis, novaGroup, nutrientLevels);
+
+                                    FragmentTransaction fragmentTransaction =  getActivity().getSupportFragmentManager().beginTransaction();
+                                    fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+
+                                    ProductDetailsFragment productDetailsFragment = new ProductDetailsFragment();
+                                    Bundle bundle = new Bundle();
+                                    bundle.putParcelable(getString(R.string.scannedProduct), Parcels.wrap(scannedProduct));
+                                    productDetailsFragment.setArguments(bundle);
+
+                                    fragmentTransaction.replace(R.id.fragment_container, productDetailsFragment);
+                                    fragmentTransaction.addToBackStack(null);
+                                    fragmentTransaction.commit();
+                                } catch (JSONException e) {
+                                    e.printStackTrace();
+                                }
+                            }
+                                }, new Response.ErrorListener() {
+                            @Override
+                            public void onErrorResponse(VolleyError error) {
+                                Log.e(TAG, "error finding product", error);
+                            }
```

Comment:
we should show a toast a user even something goes wrong. So they know what to do next